app/controllers/users/comments_controller.rb

```diff
@@ -0,0 +1,34 @@
+module Users
+  class CommentsController < Users::BaseController
+    expose :comment
+
+    def create
+      self.comment = create_comment.comment
+
+      if create_comment.success?
+        respond_with comment, location: comment.article
+      else
+        redirect_to article_path(comment.article), alert: create_comment.error
+      end
+    end
+
+    private
+
+    def authorize_resource!
+      authorize! comment
+    end
+
+    def comment_params
+      params.require(:comment).permit(:content).merge(
+        {
+          user: current_user,
+          article_id: params.require(:article_id)
```

Comment:
What do you think about this option?
1. use `parent` option of expose
2. pass `comment` to interactor

But you can leave your option if you want.

```ruby
module Users
  class CommentsController < Users::BaseController
    expose :article
    expose :comment, parent: :article

    def create
      if create_comment.success?
        respond_with comment, location: article
      else
        redirect_to article_path(article), alert: create_comment.error
      end
    end

    private

    def authorize_resource!
      authorize! comment
    end

    def comment_params
      params.require(:comment).permit(:content).merge(user: current_user)
    end

    def create_comment
      @create_comment ||= CreateComment.call(comment: comment, comment_params: comment_params)
    end
  end
end
```

Reply:
Oh, that is brilliant! 💎 
Thank you!

Reply:
I think, it's ok.